### PR TITLE
Issue #4708: fixed part of idea violations

### PR DESCRIPTION
**Pull request by Nimfadora**
Issue #4708

fixed part of [idea violations](https://nimfadora.github.io/idea/suppressed_2_part.html)

added some violations to myAllowedSuppressions:
```
                <option value="AssignmentToStaticFieldFromInstanceMethod" />
                <option value="DeprecatedIsStillUsed" />
                <option value="NoopMethodInAbstractClass" />
                <option value="UseOfSystemOutOrSystemErr" />
```
they should be commented somehow

### CODE REVIEW

**Comment by romani** [comment]
> `@SuppressWarnings("NoopMethodInAbstractClass")`

We do NOT use java annotations to suppress as it pollute the code. 
We use javadoc tag for this `@noinspection`, please use it, there are examples in code already , before usage of suppression it should be allowed at https://github.com/checkstyle/checkstyle/blob/master/config/intellij-idea-inspections.xml#L1999.

**Comment by rnveach** [comment]
> We do NOT use java annotations to suppress as it pollute the code.

@romani There is another one at https://github.com/checkstyle/checkstyle/blob/0c2abadbb7f2887f31011157d62edf40bea131c3/src/test/java/com/puppycrawl/tools/checkstyle/api/AbstractFileSetCheckTest.java#L68 .
We can use `SuppressWarningsCheck` to only allow specific suppressions.

**Comment by romani** [comment]
>We can use SuppressWarningsCheck to only allow specific suppressions

This Check only can find forbidden suppressions. I do not see easy way to keep under control as names are hard to distinguish between jdk and inspections, some inspections are lowercased, but most of them are came cased, .... May be we should target bigger part and it enough.

**Comment by Nimfadora** [comment]
@romani build fails on TC because of Deprecated API usage in pom xml, it is very weird. I think that src/main/resources and all configs in root dir should be excluded in Checkstyle Inspection Scope, what do you think?

**Comment by rnveach** [comment]
> This Check only can find forbidden suppressions.

If you make the regular expression a negative, you can turn it into a list of the only things allowed.
Example: `^((?!unchecked|deprecation).)*$` will only allow unchecked and deprecation and forbid anything else.

**Comment by romani** [comment]
@Nimfadora ,

> root dir should be excluded in Checkstyle Inspection Scope, what do you think?

if you exclude files in "Checkstyle Inspection Scope" scope, you exclude files for all validations, that is not good.
please use:
```
  <inspection_tool class="DeprecatedClassUsageInspection" enabled="true" level="WARNING" enabled_by_default="false">
    <scope name="Production" level="WARNING" enabled="true" />
  </inspection_tool>
```

**Comment by romani** [comment]
> will only allow unchecked and deprecation and forbid anything else.

good idea, applied:
https://github.com/checkstyle/checkstyle/commit/e02d6889a7474d7e2e796f1244eabcba4d4ce99d
@Nimfadora , please rebase on latest master to avoid such mistakes in future, checkstyle will guide.

File: pom.xml [hunk 1]
```diff
@@ -1,5 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 
+<!--suppress DeprecatedClassUsageInspection, DeprecatedClassUsageInspection, DeprecatedClassUsageInspection, DeprecatedClassUsageInspection, DeprecatedClassUsageInspection -->
```
Comment (romani) [inline pom.xml]: do we really need this suppression to be placed 5 times ?
It should be covered by "Production" scope usage.

Comment (Nimfadora) [inline pom.xml]: fixed

**Pull request merged**